_posts/2017-04-10-snowplow-javascript-tracker-2.8.0-released.md

```diff
@@ -0,0 +1,302 @@
+
+Previous versions of the JavaScript Tracker tracked changes to password fields
+in forms if you had form tracking enabled via:
+
+{% highlight javascript %}
+snowplow_name_here('enableFormTracking');
+{% endhighlight %}
+
+Thanks to the work of [Snowflake Analytics' Mike Robins][miike], as of this release password fields in forms won't be auto-tracked.
+
+<h2 id="contexts">4. New OptimizelyX and Parrable contexts</h2>
+
+<h3 id="parrable">4.1 Parrable context</h3>
+
+[Parrable][parrable] is a device identification platform which
+lets you idenitfy devices across browser and apps without personally
+identifiable information. This release introduces support for a Parrable context
+which will keep track of this identifier.
+
+You can enable this context when initializing the JavaScript Tracker:
+
+{% highlight js %}
+window.snowplow('newTracker', 'cf', 'd3rkrsqld9gmqf.cloudfront.net', {
+  appId: 'CFe23a',
+  platform: 'web',
+  contexts: {
+    ...
+    'parrable': true
+  }
+});
+{% endhighlight %}
+
+You can also have a look at [the JSON Schema for the Parrable context][parrable-context].
+
+There are further setup steps required to make use of Parrable data in the Snowplow pipeline; we will provide a tutorial for this soon.
+
+Many thanks to [Parrable's Chantal Galvez Omaña][chantal] for contributing the
+Parrable support to the JavaScript Tracker!
+
+<h3 id="optimizelyx">4.2 OptimizelyX context</h3>
+
+This release also brings support for [OptimizelyX][optimizelyx], which supplements the
+support for the original Optimizely service introduced in [version 2.6.0 of the Tracker](
+/blog/2016/03/03/snowplow-javascript-tracker-2.6.0-released-with-optimizely-and-augur-integration/#optimizely-integration).
+
+You can enable this context like any other:
+
+{% highlight js %}
+window.snowplow('newTracker', 'cf', 'd3rkrsqld9gmqf.cloudfront.net', {
+  appId: 'CFe23a',
+  platform: 'web',
+  contexts: {
+    ...
+    'optimizelyXSummary': true
+  }
+});
+{% endhighlight %}
+
+The schema for this context is available [in Iglu Central][optimizelyx-context].
+
+If you are planning on leveraging the context's
+`variationName`s, please untick 'Mask descriptive names in project code
+and third-party integrations' in the OptimizelyX menu -> Settings -> Privacy.
+Otherwise, all variation names will be `null`.
+
+Thanks to [Snowflake Analytics' Mike Robins][mike] for adding the support for
+OptimizelyX.
+
+<h2 id="get-data">5. Extracting key data from the tracker</h2>
+
+In an effort to give you more control, we've exposed a few properties internal
+to the tracker such as cookie names, page view IDs and domain session indices.
+
+For example, you can retrieve the complete cookie name for the domain cookie
+with through a callback to avoid problems if the tracker isn't done loading:
+
+{% highlight javascript %}
+window.snowplow_name_here(function () {
+  doSomethingWithDomainCookieName(this.getCookieName('id'));
+});
+{% endhighlight %}
```

Comment:
I think the code snippet needs to be updated because you need to reference the object created when the tracker is initialized - in the following example `cf`:

```
// Configure a tracker instance named "cf"
snowplow('newTracker', 'cf', 'd3rkrsqld9gmqf.cloudfront.net', {
	appId: 'snowplowExampleApp',
	platform: 'web'
});

// Access the tracker instance inside a callback
snowplow(function () {
	var cf = this.cf;
	var userFingerprint = cf.getUserFingerprint();
	doSomethingWith(userFingerprint);
})
```